torch/csrc/jit/codegen/cuda/scheduler/utils.cpp

```diff
@@ -2316,6 +2454,281 @@ size_t expandVectorizationToContigMergedDomains(
+          return dep_on_view_.count(v);
+        });
+
+    auto out_tvs = ir_utils::filterByType<TensorView>(e->outputs());
+
+    bool is_view_op = e->isA<ViewOp>() &&
+        std::any_of(out_tvs.begin(), out_tvs.end(), [](TensorView* out_tv) {
+                        return out_tv->hasRFactor();
+                      });
+
+    // Forward dependencies on views, assume if any input is dependent on a
+    // view all outputs are (not necessarily true with grouped reductions.)
+    if (has_view_dep_input || is_view_op) {
+      dep_on_view_.insert(e->outputs().begin(), e->outputs().end());
+    }
+
+    if (has_view_dep_input && is_view_op) {
+      // Inputs are dependent on a view, and this operation is a view so
+      // there's views that are dependent on views.
+      dep_views_ = true;
+    }
+  }
+
+ public:
+  // Return if there are any view's that have dependencies on other view
+  // operations.
+  static bool hasDependentViews(Fusion* fusion) {
+    DependentViewCheck view_check;
+    view_check.traverse(fusion);
+    return view_check.dep_views_;
+  }
+};
+
+bool hasDependentViews(Fusion* fusion) {
+  return DependentViewCheck::hasDependentViews(fusion);
+}
+
+TORCH_CUDA_CU_API bool breakIsDisjoint(std::vector<int> group_ids, int pos) {
+  if (pos < 0) {
+    pos += group_ids.size();
+  }
+  TORCH_INTERNAL_ASSERT(
+      pos >= 0 && pos <= group_ids.size(),
+      "Invalid position, size of vec is ",
+      group_ids.size(),
+      " but position is ",
+      pos);
+
+  if (pos == 0 || pos == group_ids.size()) {
+    return true;
+  }
+
+  std::unordered_set<int> left_ints;
+  for (auto i : c10::irange(pos)) {
+    left_ints.emplace(group_ids[i]);
+  }
+
+  for (auto i = pos; i < group_ids.size(); i++) {
+    if (left_ints.count(group_ids[i]) > 0) {
+      return false;
+    }
+  }
+  return true;
+}
+
+std::unordered_map<int, int> domainReorderAsRfactorMap(TensorView* tv) {
+  FusionGuard fg(tv->fusion());
+  auto transform_exprs = StmtSort::getExprs(
+      tv->fusion(),
+      {tv->domain()->domain().begin(), tv->domain()->domain().end()});
+  // simply update this vector of id's as progressing through the transformation
+  // expressions. We'll always insert the result of split in the location of the
+  // input, and insert the merge result in the position of the inner dimension.
+
+  auto reordered_ids = tv->getMaybeRFactorDomain();
+  for (const auto* expr : transform_exprs) {
+    if (const Split* split = dynamic_cast<const Split*>(expr)) {
+      auto find_it =
+          std::find(reordered_ids.begin(), reordered_ids.end(), split->in());
+      if (find_it == reordered_ids.end()) {
+        // Transformations before rfactor, ignore those.
+        continue;
+      }
+      auto pos = std::distance(reordered_ids.begin(), find_it);
+      reordered_ids.erase(reordered_ids.begin() + pos);
+      reordered_ids.insert(reordered_ids.begin() + pos, split->inner());
```

Comment:
Nit: the following has better performance
```suggestion
      reordered_ids[pos] = split->inner();
```